pkg/metrics/timer/timer.go

```diff
@@ -0,0 +1,95 @@
+package timer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	listOfStages []string
+	stageTimes   map[string]time.Duration
+	startTimes   map[string]time.Time
+}
+
+const (
+
+	// TotalTimeElapsed is a constant string value to denote total time elapsed.
+	TotalTimeElapsed = "Total"
+)
+
+var timer = NewTimer()
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer(key string) {
+	timer.StartTimer(key)
+}
+
+// StopTimer records the duration for the current stage sent as the key parameter and stores the information.
+func StopTimer(key string) {
+	timer.StopTimer(key)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	timer.LogSummary(logrus.StandardLogger())
+}
+
+// NewTimer returns a new timer that can be used to track sections and
+func NewTimer() Timer {
+	return Timer{
+		listOfStages: []string{},
+		stageTimes:   make(map[string]time.Duration),
+		startTimes:   make(map[string]time.Time),
+	}
+}
+
+// StartTimer initializes the timer object with the current timestamp information.
+func (t *Timer) StartTimer(key string) {
+	t.listOfStages = append(t.listOfStages, key)
+	t.startTimes[key] = time.Now().Round(time.Second)
+}
+
+// StopTimer records the duration for the current stage sent as the key parameter and stores the information.
+func (t *Timer) StopTimer(key string) {
+	if item, found := t.startTimes[key]; found {
+		duration := time.Now().Round(time.Second).Sub(item).Round(time.Second)
```

Comment:
https://golang.org/pkg/time/#Since

Reply:
Seems better. Changed the code to use it.